### PR TITLE
Job service

### PR DESCRIPTION
**Pull request by adam-fowler**
Restructure Job handlers to use TaskGroup and conform to ServiceLifecycle `Service`

### CODE REVIEW

File: Sources/HummingbirdJobs/JobQueueHandler.swift [hunk 1]
```diff
@@ -12,89 +12,94 @@
...
     }
 
-    /// Start queue workers
-    public func start() {
-        self.queue.onInit(on: self.eventLoop).whenComplete { _ in
-            self.workers.forEach {
-                $0.start()
+    public func run() async throws {
+        try await withGracefulShutdownHandler {
+            try await withThrowingTaskGroup(of: Void.self) { group in
```
Comment (Joannis) [inline Sources/HummingbirdJobs/JobQueueHandler.swift:36]: Shouldn't this be a discarding task group?

**Review by Joannis — APPROVED** [review]
Just one comment

File: Sources/HummingbirdJobs/JobQueueHandler.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/HummingbirdJobs/JobQueueHandler.swift:36]: A discarding task group doesn't give you any info on when a child task has finished. So I have no way to control how many tasks are running. I think a normal task group is ok as long as you are calling next() as that'll drop finished tasks off the list. I will need to check that.

Comment (adam-fowler) [reply Sources/HummingbirdJobs/JobQueueHandler.swift:36]: Going to merge this and revisit the usage to withTaskGroup

**Pull request merged**